src/mongocrypt-ctx.c

```diff
@@ -294,14 +293,23 @@ mongocrypt_ctx_t *mongocrypt_ctx_new(mongocrypt_t *crypt) {
         CLIENT_ERR("cannot create context from uninitialized crypt");
         return NULL;
     }
-    ctx_size = sizeof(_mongocrypt_ctx_encrypt_t);
-    if (sizeof(_mongocrypt_ctx_decrypt_t) > ctx_size) {
-        ctx_size = sizeof(_mongocrypt_ctx_decrypt_t);
-    }
-    if (sizeof(_mongocrypt_ctx_datakey_t) > ctx_size) {
-        ctx_size = sizeof(_mongocrypt_ctx_datakey_t);
-    }
-    ctx = bson_malloc0(ctx_size);
+
+    // Allocate enough memory for any possible context type.
+    static const size_t ctx_size = BSON_MAX(
+        sizeof(_mongocrypt_ctx_encrypt_t),
+        BSON_MAX(sizeof(_mongocrypt_ctx_decrypt_t),
+                 BSON_MAX(sizeof(_mongocrypt_ctx_datakey_t),
+                          BSON_MAX(sizeof(_mongocrypt_ctx_rewrap_many_datakey_t), sizeof(_mongocrypt_ctx_compact_t)))));
```

Comment:
```suggestion
    static const size_t ctx_size = MONGOCRYPT_CTX_ALLOC_SIZE;
```
Given suggestion in other comment to use a macro constant.